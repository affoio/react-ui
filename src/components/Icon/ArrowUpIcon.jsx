import React from 'react';
import Icon from './Icon';

const ArrowUpIcon = props => (
  <Icon {...props} width="24" height="24" viewBox="0 0 24 24">
    <path d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z" />
  </Icon>
);

export default ArrowUpIcon;
