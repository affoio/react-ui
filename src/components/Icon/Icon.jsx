import styled from 'styled-components';

const Icon = styled.svg`
  fill: ${({ color }) => color || 'currentColor'};
  ${({ hoverColor }) => (hoverColor ? `&:hover { fill: ${hoverColor};}` : '')}
`;

export default Icon;
