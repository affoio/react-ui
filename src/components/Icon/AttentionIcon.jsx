import React from 'react';
import styled from 'styled-components';
import Icon from './Icon';
import Flex from '../Grid/Flex';

const AttentionIconStyled = styled(Flex)`
  height: 24px;
  width: 24px;
`;

const AttentionIcon = props => (
  <AttentionIconStyled center middle>
    <Icon color="#FFCA00" {...props} width="22" height="19" viewBox="0 0 22 19">
      <path d="M0 19h22L11 0 0 19zm12-3h-2v-2h2v2zm0-4h-2V8h2v4z" />
    </Icon>
  </AttentionIconStyled>
);

export default AttentionIcon;
