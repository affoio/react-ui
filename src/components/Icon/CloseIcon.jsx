import React from 'react';
import styled from 'styled-components';
import Icon from './Icon';
import Flex from '../Grid/Flex';

const CloseIconStyled = styled(Flex)`
  height: 16px;
  width: 16px;
`;

const CloseIcon = props => (
  <CloseIconStyled center middle>
    <Icon {...props} width="10" height="10" viewBox="0 0 10 10">
      {/* eslint-disable-next-line max-len */}
      <path d="M9.667 1.273l-.94-.94L5 4.06 1.273.333l-.94.94L4.06 5 .333 8.727l.94.94L5 5.94l3.727 3.727.94-.94L5.94 5z" />
    </Icon>
  </CloseIconStyled>
);

export default CloseIcon;
