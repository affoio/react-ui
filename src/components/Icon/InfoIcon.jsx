import React from 'react';
import styled from 'styled-components';
import Icon from './Icon';
import Flex from '../Grid/Flex';

const InfoIconStyled = styled(Flex)`
  height: 24px;
  width: 24px;
`;

const InfoIcon = props => (
  <InfoIconStyled center middle>
    <Icon color="#4718B4" {...props} width="20" height="20" viewBox="0 0 20 20">
      <path d="M10 20c5.52 0 10-4.48 10-10S15.52 0 10 0 0 4.48 0 10s4.48 10 10 10zM9 5h2v2H9V5zm0 4h2v6H9V9z" />
    </Icon>
  </InfoIconStyled>
);

export default InfoIcon;
