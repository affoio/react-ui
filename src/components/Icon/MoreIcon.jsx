import React from 'react';
import Icon from './Icon';
import colors from '../../colors/index';

const MoreIcon = props => (
  <Icon color={colors.black[400]} {...props} width="24" height="24" viewBox="0 0 6 24">
    <g fillRule="evenodd" transform="translate(0, 4)">
      {/* eslint-disable-next-line max-len */}
      <path d="M2 4c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2C.9 6 0 6.9 0 8s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z" />
    </g>
  </Icon>
);

export default MoreIcon;
