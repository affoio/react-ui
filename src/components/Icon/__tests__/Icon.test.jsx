import React from 'react';
import { render } from 'enzyme';
import Icon from '../Icon';

describe('<Icon>', () => {
  it('renders correctly', () => {
    const tree = render(<Icon />);

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with color', () => {
    const tree = render(<Icon color="red" />);

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with hoverColor', () => {
    const tree = render(<Icon hoverColor="red" />);

    expect(tree).toMatchSnapshot();
  });
});
