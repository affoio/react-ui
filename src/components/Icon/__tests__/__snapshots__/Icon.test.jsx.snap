// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon> renders correctly 1`] = `
.c0 {
  fill: currentColor;
}

<svg
  class="c0"
/>
`;

exports[`<Icon> renders correctly with color 1`] = `
.c0 {
  fill: red;
}

<svg
  class="c0"
  color="red"
/>
`;

exports[`<Icon> renders correctly with hoverColor 1`] = `
.c0 {
  fill: currentColor;
}

.c0:hover {
  fill: red;
}

<svg
  class="c0"
/>
`;
